VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Activities/Productizer/Operations/BasicInformation/UpdatePersonBasicInformation.cs

```diff
@@ -0,0 +1,78 @@
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Swashbuckle.AspNetCore.Annotations;
+using VirtualFinland.UserAPI.Data;
+using VirtualFinland.UserAPI.Helpers.Swagger;
+
+namespace VirtualFinland.UserAPI.Activities.Productizer.Operations.BasicInformation;
+
+public static class UpdatePersonBasicInformation
+{
+    public class Command : IRequest<UpdatePersonBasicInformationResponse>
+    {
+        public Command(string? givenName, string? lastName, string email, string? phoneNumber, string residency)
+        {
+            GivenName = givenName;
+            LastName = lastName;
+            Email = email;
+            PhoneNumber = phoneNumber;
+            Residency = residency;
+        }
+
+        [SwaggerIgnore]
+        public Guid? UserId { get; set; }
+
+        public string? GivenName { get; }
+        public string? LastName { get; }
+        public string Email { get; }
+        public string? PhoneNumber { get; }
+        public string? Residency { get; }
+
+        public void SetAuth(Guid? userDatabaseId)
+        {
+            UserId = userDatabaseId;
+        }
+    }
+
+    public class Handler : IRequestHandler<Command, UpdatePersonBasicInformationResponse>
+    {
+        private readonly UsersDbContext _context;
+
+        public Handler(UsersDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<UpdatePersonBasicInformationResponse> Handle(Command request,
+            CancellationToken cancellationToken)
+        {
+            var person = await _context.Persons.SingleAsync(p => p.Id == request.UserId, cancellationToken);
+
+            person.GivenName = request.GivenName;
+            person.LastName = request.LastName;
+            person.Email = request.Email;
+            person.PhoneNumber = request.PhoneNumber;
+            person.ResidencyCode = request.Residency; // TODO: This should probably just be string
+
+            await _context.SaveChangesAsync(cancellationToken);
```

Comment:
Tässä jos yrittää tallentaa vääränlaista tietoa (esim. residencyksi 'abcd') niin tietokanta kieltäytyy, mutta koska virhettä ei kaapata tulee siitä oletusvirheviesti 500 internal server error. Koska kyse on kuitenkin vääränlaisesta inputista niin olettaisin saavani validaatiovirheen 422. 

En ihan tiedä onko postgres-ajurin virheviesti sellainen että siitä saisi järkevän kohdennuksen sille että mikä kenttä meni pieleen, mutta näyttäisi että ainakin tuosta `Npgsql.PostgresException` -virheestä saisi huomattua että kyse data-virheestä jos error-koodi kuuluu alkaa numeroilla 22.. eli kuuluu luokkaan `Class 22 — Data Exception` (https://www.postgresql.org/docs/current/errcodes-appendix.html) 

Tämän ja kaltaiset possun kertomat data-virheet voisi siis kaapata ja lähettää eteenpäin esim genereenisenä ValidationError-exceptionina, jonka sit saisi kaapattua tuolla `ErrorHandlerMiddleware` ja siellä laitettua oikean statuskoodin.  Tai ehkä jopa tehdä errror handleriin yleisellä tasolla käsittely postgres:n data-virheille.

Reply:
Tuollainen virheenkäsittely olisi varmastikin hyvä, mutta esim näissä productizer endpointeissa se pyyntö jää käsittääkseni kiinni jo testbedissä, mikäli data ei ole validia. 

Tähän voisi kehittää jotain järkevää yleiskäyttöistä ratakaisua, jota en nyt itse ihan saanut tehtyä. Yksi ongelma on, että C# ei voi käyttää enumeita suoraan mitenkään fiksusti, koska esim 01 ei ole sallitu nimi Enumille, ja arvot ovat integereitä. Muutenkin jos voitaitiin geenroida POCO:t testbedin määrittelystä oikeilla validaatiosäännöillä, vähentäisi se työn ja virheiden mahdollisuuksien määrää huomattavasti.

Reply:
Juups, se on kyllä jo hyvä suoja tähän käyttötarkoitukseen että testbedin päässä tehdään varmistukset.